### ISSUE TITLE
Current master completely fails to build (MacOS)

### ISSUE THREAD

**Issue by mouse07410** [open-quantum-safe/liboqs#955]
MacOS Catalina 10.15.7, Xcode-12.4

CMake config command:
```
$ env RUN_TESTS_PARALLEL_UPPER=FALSE cmake -GNinja .. -DOQS_USE_OPENSSL=ON -DOQS_DIST_BUILD=OFF -DBUILD_SHARED_LIBS=ON -DCMAKE_INSTALL_PREFIX=/opt/local -DOQS_KEM_DEFAULT="OQS_KEM_alg_ntru_hps4096821" -DOQS_SIG_DEFAULT="OQS_SIG_alg_falcon_1024" -DCMAKE_BUILD_TYPE=Release

-- The C compiler identification is Clang 11.1.0
-- The ASM compiler identification is Clang
-- Found assembler: /opt/local/bin/clang
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /opt/local/bin/clang - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE  
-- Found OpenSSL: /opt/local/lib/libcrypto.dylib (found suitable version "1.1.1k", minimum required is "1.1.1")  
-- Looking for OPENSSL_NO_EC2M
-- Looking for OPENSSL_NO_EC2M - not found
-- Found Doxygen: /opt/local/bin/doxygen (found version "1.9.1") found components: doxygen dot 
-- Configuring done
-- Generating done
-- Build files have been written to: /Users/ur20980/src/liboqs/build

CMake completed configuration for Ninja
```

Build failure:
```
[1445/2244] Building ASM object src/common/sha3/xkcp_...1600_avx2.dir/KeccakP-1600/avx2/KeccakP-1600-AVX2.s.o
FAILED: src/common/sha3/xkcp_low/CMakeFiles/xkcp_low_keccakp_1600_avx2.dir/KeccakP-1600/avx2/KeccakP-1600-AVX2.s.o 
/opt/local/bin/clang  -Iinclude -I../src/common/sha3/xkcp_low/KeccakP-1600/avx2 -O3 -DNDEBUG -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk -mmacosx-version-min=10.15 -fPIC -march=native -Werror -Wall -Wextra -Wpedantic -Wno-unused-command-line-argument -O3 -fomit-frame-pointer -Wbad-function-cast -Wcast-qual -Wnarrowing -Wconversion -mavx2 -MD -MT src/common/sha3/xkcp_low/CMakeFiles/xkcp_low_keccakp_1600_avx2.dir/KeccakP-1600/avx2/KeccakP-1600-AVX2.s.o -MF src/common/sha3/xkcp_low/CMakeFiles/xkcp_low_keccakp_1600_avx2.dir/KeccakP-1600/avx2/KeccakP-1600-AVX2.s.o.d -o src/common/sha3/xkcp_low/CMakeFiles/xkcp_low_keccakp_1600_avx2.dir/KeccakP-1600/avx2/KeccakP-1600-AVX2.s.o -c ../src/common/sha3/xkcp_low/KeccakP-1600/avx2/KeccakP-1600-AVX2.s
../src/common/sha3/xkcp_low/KeccakP-1600/avx2/KeccakP-1600-AVX2.s:29:1: error: unknown directive
.type KeccakP1600_Initialize,@function
^
../src/common/sha3/xkcp_low/KeccakP-1600/avx2/KeccakP-1600-AVX2.s:44:1: error: unknown directive
.size KeccakP1600_Initialize,.-KeccakP1600_Initialize
^
../src/common/sha3/xkcp_low/KeccakP-1600/avx2/KeccakP-1600-AVX2.s:57:1: error: unknown directive
.type KeccakP1600_AddByte,@function
^
../src/common/sha3/xkcp_low/KeccakP-1600/avx2/KeccakP-1600-AVX2.s:72:1: error: unknown directive
.size KeccakP1600_AddByte,.-KeccakP1600_AddByte
^
../src/common/sha3/xkcp_low/KeccakP-1600/avx2/KeccakP-1600-AVX2.s:85:1: error: unknown directive
.type KeccakP1600_AddBytes,@function
^
../src/common/sha3/xkcp_low/KeccakP-1600/avx2/KeccakP-1600-AVX2.s:139:1: error: unknown directive
.size KeccakP1600_AddBytes,.-KeccakP1600_AddBytes
^
../src/common/sha3/xkcp_low/KeccakP-1600/avx2/KeccakP-1600-AVX2.s:152:1: error: unknown directive
.type KeccakP1600_OverwriteBytes,@function
^
../src/common/sha3/xkcp_low/KeccakP-1600/avx2/KeccakP-1600-AVX2.s:206:1: error: unknown directive
.size KeccakP1600_OverwriteBytes,.-KeccakP1600_OverwriteBytes
^
../src/common/sha3/xkcp_low/KeccakP-1600/avx2/KeccakP-1600-AVX2.s:219:1: error: unknown directive
.type KeccakP1600_OverwriteWithZeroes,@function
^
../src/common/sha3/xkcp_low/KeccakP-1600/avx2/KeccakP-1600-AVX2.s:249:1: error: unknown directive
.size KeccakP1600_OverwriteWithZeroes,.-KeccakP1600_OverwriteWithZeroes
^
../src/common/sha3/xkcp_low/KeccakP-1600/avx2/KeccakP-1600-AVX2.s:262:1: error: unknown directive
.type KeccakP1600_ExtractBytes,@function
^
../src/common/sha3/xkcp_low/KeccakP-1600/avx2/KeccakP-1600-AVX2.s:319:1: error: unknown directive
.size KeccakP1600_ExtractBytes,.-KeccakP1600_ExtractBytes
^
../src/common/sha3/xkcp_low/KeccakP-1600/avx2/KeccakP-1600-AVX2.s:332:1: error: unknown directive
.type KeccakP1600_ExtractAndAddBytes,@function
^
../src/common/sha3/xkcp_low/KeccakP-1600/avx2/KeccakP-1600-AVX2.s:397:1: error: unknown directive
.size KeccakP1600_ExtractAndAddBytes,.-KeccakP1600_ExtractAndAddBytes
^
../src/common/sha3/xkcp_low/KeccakP-1600/avx2/KeccakP-1600-AVX2.s:406:1: error: unknown directive
.type __KeccakF1600,@function
^
../src/common/sha3/xkcp_low/KeccakP-1600/avx2/KeccakP-1600-AVX2.s:549:1: error: unknown directive
.size __KeccakF1600,.-__KeccakF1600
^
../src/common/sha3/xkcp_low/KeccakP-1600/avx2/KeccakP-1600-AVX2.s:559:1: error: unknown directive
.type KeccakP1600_Permute_24rounds,@function
^
../src/common/sha3/xkcp_low/KeccakP-1600/avx2/KeccakP-1600-AVX2.s:588:1: error: unknown directive
.size KeccakP1600_Permute_24rounds,.-KeccakP1600_Permute_24rounds
^
../src/common/sha3/xkcp_low/KeccakP-1600/avx2/KeccakP-1600-AVX2.s:596:1: error: unknown directive
.type KeccakP1600_Permute_12rounds,@function
^
../src/common/sha3/xkcp_low/KeccakP-1600/avx2/KeccakP-1600-AVX2.s:625:1: error: unknown directive
.size KeccakP1600_Permute_12rounds,.-KeccakP1600_Permute_12rounds
^
../src/common/sha3/xkcp_low/KeccakP-1600/avx2/KeccakP-1600-AVX2.s:633:1: error: unknown directive
.type KeccakP1600_Permute_Nrounds,@function
^
../src/common/sha3/xkcp_low/KeccakP-1600/avx2/KeccakP-1600-AVX2.s:664:1: error: unknown directive
.size KeccakP1600_Permute_Nrounds,.-KeccakP1600_Permute_Nrounds
^
../src/common/sha3/xkcp_low/KeccakP-1600/avx2/KeccakP-1600-AVX2.s:677:1: error: unknown directive
.type KeccakF1600_FastLoop_Absorb,@function
^
../src/common/sha3/xkcp_low/KeccakP-1600/avx2/KeccakP-1600-AVX2.s:845:1: error: unknown directive
.size KeccakF1600_FastLoop_Absorb,.-KeccakF1600_FastLoop_Absorb
^
../src/common/sha3/xkcp_low/KeccakP-1600/avx2/KeccakP-1600-AVX2.s:858:1: error: unknown directive
.type KeccakP1600_12rounds_FastLoop_Absorb,@function
^
../src/common/sha3/xkcp_low/KeccakP-1600/avx2/KeccakP-1600-AVX2.s:1026:1: error: unknown directive
.size KeccakP1600_12rounds_FastLoop_Absorb,.-KeccakP1600_12rounds_FastLoop_Absorb
^
[1466/2244] Building C object src/common/sha3/xkcp_lo...ccakP-1600times4/avx2/KeccakP-1600-times4-SIMD256.c.o
ninja: build stopped: subcommand failed.
```

I seem to recall that we've been here already - looks like a regression.

**Comment by baentsch** [comment]
Looks like our CI doesn't have AVX2 to confirm the fix in #956: @mouse07410 could you please confirm that fix resolves this issue for you?

**Comment by mouse07410** [comment]
@baentsch yes I confirm - with #956 merged, it resolved this issue. Thanks!
